homeassistant/components/renault/binary_sensor.py

```diff
@@ -59,10 +61,39 @@ class RenaultBinarySensor(
     @property
     def is_on(self) -> bool | None:
         """Return true if the binary sensor is on."""
-        if (data := self._get_data_attr(self.entity_description.on_key)) is None:
-            return None
 
-        return data == self.entity_description.on_value
+        if self.entity_description.value_lambda is not None:
+            return self.entity_description.value_lambda(self)
+        if self.entity_description.on_key is not None:
+            if (data := self._get_data_attr(self.entity_description.on_key)) is None:
+                return None
+
+            return data == self.entity_description.on_value
+        return None
+
+
+def _plugged_in_value_lambda(self: RenaultBinarySensor) -> bool | None:
+    """Return true if the vehicle is plugged in."""
+
+    data = self.coordinator.data
+    plug_status = data.get_plug_status() if data else None
+
+    if plug_status is None:
```

Comment:
If you invert the check, then we can un-indent the code below
```suggestion
    if plug_status is not None:
        return plug_status == PlugState.PLUGGED
```

Reply:
doing a slight different change

Reply:
OK for me... but can we keep 100% test coverage?